shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -1349,4 +1505,327 @@ public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOE
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+        }
+
+        public static WireCommand readFrom(DataInput in, int length) throws IOException {
+            long requestId = in.readLong();
+            String segment = in.readUTF();
+            return new TableKeysRemoved(requestId, segment);
+        }
+    }
+
+    @Data
+    public static final class ReadTable implements Request, WireCommand {
+
+        final WireCommandType type = WireCommandType.READ_TABLE;
+        final long requestId;
+        final String segment;
+        final String delegationToken;
+        final List<TableKey> keys; // the version of the key is always set to io.pravega.segmentstore.contracts.tables.TableKey.NO_VERSION
+
+        @Override
+        public void process(RequestProcessor cp) {
+            cp.readTable(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+            out.writeUTF(delegationToken);
+            out.writeInt(keys.size());
+            for (TableKey key : keys) {
+                key.writeFields(out);
+            }
+        }
+
+        public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOException {
+            long requestId = in.readLong();
+            String segment = in.readUTF();
+            String delegationToken = in.readUTF();
+            int numberOfKeys = in.readInt();
+            List<TableKey> keys = new ArrayList<>(numberOfKeys);
+            for (int i = 0; i < numberOfKeys; i++) {
+                keys.add((TableKey) TableKey.readFrom(in, in.available()));
+            }
+            return new ReadTable(requestId, segment, delegationToken, keys);
+        }
+    }
+
+    @Data
+    public static final class TableRead implements Reply, WireCommand {
+        final WireCommandType type = WireCommandType.TABLE_READ;
+        final long requestId;
+        final String segment;
+        final TableEntries entries;
+
+        @Override
+        public void process(ReplyProcessor cp) {
+            cp.tableRead(this);
+        }
+
+        @Override
+        public void writeFields(DataOutput out) throws IOException {
+            out.writeLong(requestId);
+            out.writeUTF(segment);
+            entries.writeFields(out);
+        }
+
+        public static WireCommand readFrom(ByteBufInputStream in, int length) throws IOException {
+            long requestId = in.readLong();
+            String segment = in.readUTF();
+            TableEntries entries = (TableEntries) TableEntries.readFrom(in, in.available());
+            return new TableRead(requestId, segment, entries);
+        }
+    }
+
+    @Data
+    public static final class TableEntries implements WireCommand {
+        final WireCommandType type = WireCommandType.TABLE_ENTRIES;
```

Comment:
It looks to me like TableEntries is just a convenience for embedding into other commands. In which case it doesn't need it's own WireCommandType because it's not actually used.

Reply:
yes, it is a convenience class  and should not need its own WireCommandType. Removed it.